### ISSUE TITLE
Switch or at least support code listings in LaTeX using the `linstings` package instead of `minted`

### ISSUE THREAD

**Issue by BambOoxX** [JuliaDocs/Documenter.jl#2309]
Hi, I'm currently trying to build a PDF version of the documentation of one of my packages, and unfortunately, this has been really painful due to `minted` and `pygments`. I'm pretty sure there is a very good reason for using this package, but unfortunately, because of the external dependency, I simply cannot use the PDF output. 

I was wondering if using `listings` instead of `minted` was ever considered ? I know that `listings` does not officially support `julia` as a language, but this repo contains a language specification for `julia`: https://github.com/wg030/jlcode

**Comment by mortenpi** [comment]
I'm not opposed to using `listings` instead of `minted`, but what issues are you actually running into with `minted`/Pygments?

**Comment by BambOoxX** [comment]
I've had a lot of trouble with `minted` finding `pygmentize` because of `PATH` conflicts. I know it is related to my specific company setup, nevertheless it's quite cumbersome to debug. And now that it works I get an errro in a `pygtex` file saying.
> ! Undefined control sequence.
\PYG #1#2->\FV@PYG

My idea is more generally that people that will run `minted` will require a python install, which is still not everybody (at least on windows), and that this could be solved by purely relying on LaTeX, thus avoiding an additional dependency compared to julia.

**Comment by mortenpi** [comment]
I guess [Docker](https://documenter.juliadocs.org/stable/man/other-formats/#Compiling-using-docker-image) is also out of the question for you? You could also [try Tectonic though](https://documenter.juliadocs.org/stable/man/other-formats/#Compiling-using-Tectonic), but I am not sure if it handles the Python / pygments installation for you or not.

**Comment by BambOoxX** [comment]
Yes, unfortunately, both Docker and Tectonic are out of my scope. 
[Auditing the code](https://github.com/search?q=repo%3AJuliaDocs%2FDocumenter.jl%20minted&type=code), there does not seem to be so many references to `minted`, so I could try and propose a PR for `lsitings`, whenever I have some time. 
I figuring this would result in an additional opt-in kwarg for `LaTexWriter`.

**Comment by mortenpi** [comment]
Yeah, let's make this an opt-in option via a kwarg.

**Comment by odow** [comment]
I haven't benchmarked, but I think the `minted` package is one cause of the excessively large compilation times for the JuMP documentation, so I would be in favor of this. At the very least, I should check.

**Comment by BambOoxX** [comment]
@mortenpi I'm wondering what would be the correct way to configure https://github.com/JuliaDocs/Documenter.jl/blob/77f0bdd7c742fc7d7ed91c6b0ab6582f14e33e81/src/latex/LaTeXWriter.jl#L490-L519 with an opt-in option to `LaTeX(...)`.

I've added an option with
```julia
struct LaTeX <: Documenter.Writer
    platform::String
    version::String
    tectonic::Union{Cmd,String,Nothing}
    code_listings::String
    function LaTeX(;
            platform = "native",
            version  = get(ENV, "TRAVIS_TAG", ""),
            tectonic = nothing,
            code_listings = "minted")
        code_listings ∈ ("minted", "listings") || throw(ArgumentError("unknown code formatting package: $platform"))
        platform ∈ ("native", "tectonic", "docker", "none") || throw(ArgumentError("unknown platform: $platform"))
        return new(platform, string(version), tectonic, code_listings)
    end
end
```
but since the `latex(...)` calls are nested, I do not want to touch the signature too much. Would you have a suggestion ? 
Some `latex(...)` use `setting` as an option, but I preferred to have you opinion on this first. 

Side note, `tex` is currently detected with https://github.com/JuliaDocs/Documenter.jl/blob/77f0bdd7c742fc7d7ed91c6b0ab6582f14e33e81/src/latex/LaTeXWriter.jl#L90
While `latexmk` may be used by a lot of people, I think detecting a `tex` distro with that is a bit restrictive, though I guess you impose this due to the `platform` setting. If you are insterested, I think I can propose an `arara`-based equivalent compilation chain.

**Comment by BambOoxX** [comment]
@mortenpi I've started really working on this, and I was wondering what the pdf output should look like. Would you mind sharing it (I can't generate it).
